internal/validation/suite.go

```
@@ -11,25 +10,23 @@ import (
 )
 
 type ProviderConfig struct {
-	ProviderName  string
-	EnvVarName    string
-	ClientFactory func(apiKey string) v1.CloudClient
+	ProviderName string
+	Credential   v1.CloudCredential
 }
 
 func RunValidationSuite(t *testing.T, config ProviderConfig) {
 	if testing.Short() {
 		t.Skip("Skipping validation tests in short mode")
 	}
 
-	apiKey := os.Getenv(config.EnvVarName)
-	if apiKey == "" {
-		t.Skipf("%s not set, skipping %s validation tests", config.EnvVarName, config.ProviderName)
-	}
-
-	client := config.ClientFactory(apiKey)
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
 	defer cancel()
 
+	client, err := config.Credential.MakeClient(ctx, "")
+	if err != nil {
+		t.Skipf("Failed to create client for %s: %v", config.ProviderName, err)
```

we should fail here instead of skip